docs/dev/Kibana-Reporting-Design-Proposal.md

```diff
@@ -20,43 +21,111 @@ Kibana Reports for Open Distro allows ‘Report Owner’ (engineers, including b
+1. As a user, I should see Name, Type, Sender, Recepient(s), Source, Last updated, State, Download on the `Reports` list.
+1. As a user, I should be able to filter my reports list by sender, receiver, status, and type.
+1. As a user, I should be able to search my reports by report name.
+1. As a user, I should be able to view a report details page for any report on the list.
+1. As a user, I should see my reports ordered in reverse chronological order by default.
+1. As a user, I should be able to download the default file format from the reports list.
+1. As a user, I should be able to see a list of reports for the past 100 days.
+1. As a user, archived reports should not be visible by default (the default filter should exclude archived reports) (p2)
+
+
+#### Report definitions
+1. As a user, I should see Name, Type, Owner, Source, Last updated, Details, Status (active/disabled) on the `Report definitions` list.
+1. As a user, I should be able to view, edit and delete a report definition.
+1. As a user, I should be able to enable/disable a report definition
+
+### Create, View, Edit
+#### Create report definition
+##### Report settings
+1. As a user, I should be able to set the report name.
+1. As a user, I should be able to set an optional report descritption.
+1. As a user, I should be able to select a report source type (Dashboard, Visualization, Saved Search)
+1. As a user, I should be able to select a specific source (Dashboard, Visualization, or Saved search, contextually).
+1. As a user, I should be able to select the default file format for my report, contextual to my selected source (PDF, PNG, CSV, Excel, etc)
+1. As a user, I should be able to add a `Header` and `Footer` to PDF and PNG reports.
+    1. As a user, I should see Report title and a sample paragraph in the `header` by Default.
+    1. As a user, I should see date, time and report name in the `footer` by Default.
+1. As a user, I should be able to select which charts/visualizations I can include in a PDF or PNG report. (p2)
+
+##### Report trigger
+1. As a user, I should be able to select a report trigger type that belongs to one of two categories: `Schedule`, and `Alert`.
+1. As a user, when I select `Schedule`, I should be able to select a request time of now, future date, recurring, or custom cron expression (similar to alerting).
+    1. As a user, if I select `now`, I should be able to generate a report immediately after saving the report definition.
+    1. As a user, if I select `future date`, I should be able to configure the date and time the report will be generated.
+    1. As a user, if I select `recurring`, I should be able to configure a recurring schedule.
+    1. As a user, if I select `custom cron`, I should be able to configure a cron-based schedule.
+    1. As a user, if I select `recurring` or `custom cron`, I should be able to configure an end date.
+1. As a user, when I select `Alert`, I should be able to select the alert a report gets triggered from.
```

Comment:
Where to select the alerts from? What generates the list of selectable alerts?

Reply:
These will come from the Alerting plugin. In the alerting plugin, we can create monitors, and those monitors have triggers. We need to get a list of available monitors, and once a monitor has been selected, we need to get a list of available triggers to use as a report trigger target.